dashboard/src/components/OperatorNew/OperatorNew.test.tsx

```diff
@@ -0,0 +1,124 @@
+import { shallow } from "enzyme";
+import * as React from "react";
+import { NotFoundError } from "../../shared/types";
+import { wait } from "../../shared/utils";
+import UnexpectedErrorPage from "../ErrorAlert/UnexpectedErrorAlert";
+import OperatorNew from "./OperatorNew";
+
+const defaultProps = {
+  operatorName: "foo",
+  getOperator: jest.fn(),
+  isFetching: false,
+  namespace: "kubeapps",
+  push: jest.fn(),
+  createOperator: jest.fn(),
+};
+
+const defaultOperator = {
+  metadata: {
+    name: "foo",
+    namespace: "kubeapps",
+  },
+  status: {
+    provider: {
+      name: "Kubeapps",
+    },
+    defaultChannel: "beta",
+    channels: [
+      {
+        name: "beta",
+        currentCSV: "foo.1.0.0",
+        currentCSVDesc: {
+          displayName: "Foo",
+          version: "1.0.0",
+          description: "this is a testing operator",
+          annotations: {
+            capabilities: "Basic Install",
+            repository: "github.com/kubeapps/kubeapps",
+            containerImage: "kubeapps/kubeapps",
+            createdAt: "one day",
+          },
+          installModes: [],
+        },
+      },
+    ],
+  },
+} as any;
+
+it("calls getOperator when mounting the component", () => {
+  const getOperator = jest.fn();
+  shallow(<OperatorNew {...defaultProps} getOperator={getOperator} />);
+  expect(getOperator).toHaveBeenCalledWith(defaultProps.namespace, defaultProps.operatorName);
+});
+
+it("calls getOperator when changing the namespace the component", () => {
+  const getOperator = jest.fn();
+  const wrapper = shallow(<OperatorNew {...defaultProps} getOperator={getOperator} />);
+  expect(getOperator).toHaveBeenCalledWith(defaultProps.namespace, defaultProps.operatorName);
+  wrapper.setProps({ namespace: "foo" });
+  expect(getOperator).toHaveBeenCalledWith("foo", defaultProps.operatorName);
+});
+
+it("parses the default channel when receiving the operator", () => {
```

Comment:
Is it implicit here that the 0th (first) channel is the default one? (if so, clearer as "parses the first channel when receiving the operator")